lib/tests/test_rewrite.rs

```diff
@@ -647,13 +647,14 @@ fn test_rebase_descendants_multiple_sideways() {
 }
 
 #[test]
+#[should_panic(expected = "cycle detected")]
 fn test_rebase_descendants_multiple_swap() {
     let settings = testutils::user_settings();
     let test_repo = TestRepo::init();
     let repo = &test_repo.repo;
 
     // Commit B was replaced by commit D. Commit D was replaced by commit B.
-    // Commit C and commit E should swap places.
+    // This results in an infinite loop and a panic
```

Comment:
IIRC, my thinking behind this test was that if we infer from a remote that two commits swapped places, then we should swap the descendants. But we don't currently make that inference from two branches swapping place or anything else, and I feel like we probably *shouldn't* make that inference. What do you think?

Reply:
There might be more to think about, but if we absolutely needed such behavior, we could

- remember what children D has
- rebase the children of B to D.
- rebase the previously-existing children of D to B
- rebase_descendants

In each case, the rebase takes a commit to a commit with an id that never existed before.

One conclusion from this is that we would probably want to document the restriction on cycles in `DescendantRebaser::new`.

Reply:
Since we don't have the need right now, feel free to drop support for it. Hopefully we will never need that functionality :) (and to be clear, I don't think we will - I'm not being sarcastic here)